### PR TITLE
Add CoNLL2000Checking Mocked Unit Test

### PR DESCRIPTION
**Pull request by VirgileHlav**
Reference Issue: #1493

## Summary
- Added mocked unit test for CoNLL2000Checking dataset 

## Test
- ` pytest test/datasets/test_conll2000chunking.py`

### CODE REVIEW

File: test/datasets/test_conll2000chunking.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                seed += 1
+
+
+    # create gz file from dataset folder
+    for file_name in ("train.txt", "test.txt"):
+        compressed_dataset_path = os.path.join(temp_dataset_dir, f"{file_name}.gz")
+        with gzip.open(compressed_dataset_path, "wb") as gz_file:
+            txt_file = os.path.join(temp_dataset_dir, file_name)
+            with open(txt_file, "rb") as file_in:
+                gz_file.writelines(file_in)
```
Comment (Nayef211) [inline test/datasets/test_conll2000chunking.py:50]: Is there any way to simplify this? Can we add the entire folder to the archive like we've been doing for tar archives?

**Review by Nayef211 — COMMENTED** [review]
It looks the like the website that [connll2000 dataset](https://www.clips.uantwerpen.be/conll2000/chunking/train.txt.gz) is hosted on is down right now. Let's wait until it comes back up so we can verify that the mock data matches the data in the dataset before we merge this PR.

File: test/datasets/test_conll2000chunking.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+from ..common.case_utils import TempDirMixin, zip_equal
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def _get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "CoNLL2000Chunking")
+    temp_dataset_dir = os.path.join(base_dir, "conll2000chunking")
```
[the remark this answers is not in the document]
Comment (VirgileHlav) [reply test/datasets/test_conll2000chunking.py:20]: Given the [current loading implementation](https://github.com/pytorch/text/blob/main/torchtext/datasets/conll2000chunking.py#L49), we need a folder entitled `conll2000chunking` in order for the loading to WAI.

File: test/datasets/test_conll2000chunking.py (hunk 1 above)
Comment (VirgileHlav) [reply test/datasets/test_conll2000chunking.py]: I looked into it and there does not seem to be a way to use `gzip` to compress an entire folder. I simplified the logic for readability.

File: test/datasets/test_conll2000chunking.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (erip) [reply test/datasets/test_conll2000chunking.py:20]: I don't... It seems like an artifact of the original dataset impl, though.

[the remark this answers is not in the document]
Comment (Nayef211) [reply test/datasets/test_conll2000chunking.py:20]: > conll2000chunking

Once https://github.com/pytorch/text/pull/1590 has been merged in, you should no longer need this intermediate folder. This PR should be good to be merged once this is fixed 😄

[the remark this answers is not in the document]
Comment (parmeet) [reply test/datasets/test_conll2000chunking.py:20]: Yes, that's right @erip. The issue is with previous implementation. Looks like @Nayef211  is taking care of it in this PR https://github.com/pytorch/text/pull/1590 along the caching issue.

**Pull request merged**